### PR TITLE
fix: use standalone tests' exit code

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

increase stability and lower false negative rate

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
📚 Documentation preview 📚: https://pytorch-lightning--20430.org.readthedocs.build/en/20430/

cc @borda

### CODE REVIEW

File: tests/run_standalone_tests.sh [hunk 1]
```diff
@@ -19,79 +19,100 @@ set -e
...
-for pid in ${pids[*]}; do wait $pid; done
-show_batched_output
 
 # echo test report
 printf '=%.s' {1..80}
 printf "\n$report"
 printf '=%.s' {1..80}
 printf '\n'
+
+# exit with the worse test result
```
Comment (lantiga) [inline tests/run_standalone_tests.sh:117]:
```suggestion
# exit with the worst test result
```

File: tests/run_standalone_tests.sh [hunk 2]
```diff
@@ -19,79 +19,100 @@ set -e
 # It can be set through the env variable PL_STANDALONE_TESTS_BATCH_SIZE and defaults to 6 if not set
 test_batch_size="${PL_STANDALONE_TESTS_BATCH_SIZE:-6}"
 source="${PL_STANDALONE_TESTS_SOURCE:-"lightning"}"
+# this is the directory where the tests are located
+test_dir=$1 # parse the first argument
+COLLECTED_TESTS_FILE="collected_tests.txt"
+
+#cd ${test_dir}
```
Comment (lantiga) [inline tests/run_standalone_tests.sh:26]: Remove comment or actually `cd` into it?

**Review by lantiga — APPROVED** [review]
Looks good 🤞

**Comment by t-vi** [comment]
Supergood, this should get the most flaky bit of the CI under control! Thank you @Borda

**Review by justusschock — APPROVED** [review]

File: tests/run_standalone_tests.sh (hunk 2 above)
Comment (Borda) [reply tests/run_standalone_tests.sh:26]:
```suggestion
```

**Pull request merged**